src/japi_pushsrv.c

```diff
@@ -314,21 +344,42 @@ japi_pushsrv_context* japi_pushsrv_register(japi_context* ctx, const char* pushs
 	return psc;
 }
 
-/*
- * Iterate through push service and unsubscribe & free memory for all clients
+/* 
+ * Remove push service context from japi context, unsubscribe for all clients and free memory
  */
-int japi_pushsrv_destroy(japi_pushsrv_context *psc)
+int japi_pushsrv_destroy(japi_context *ctx, japi_pushsrv_context *psc)
 {
+	japi_pushsrv_context *psc_iter, *psc_prev, *psc_next;
 	japi_client *client, *client_next;
 
+	assert(ctx != NULL);
+
 	if (psc == NULL) {
-		fprintf(stderr,"ERROR: push service context is NULL");
+		fprintf(stderr,"ERROR: push service context is NULL\n");
 		return -1;
 	}
 
-	client = psc->clients;
+	/* clean up linked list in ctx->push_service */
+	psc_prev = NULL;
+	psc_iter = ctx->push_services;
+
+	while (psc_iter != NULL) {
+		psc_next = psc_iter->next;
+		if (psc_iter == psc) {
```

Comment:
May be return an error if the push service was not found?

Reply:
If the push service is not found at this point, it was already removed and this part just does nothing. It's good to anyways clean up connected clients, so no return.